orca-mine/src/main/groovy/com/netflix/spinnaker/orca/mine/pipeline/AcaTaskStage.groovy

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2015 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca.mine.pipeline
+
+import com.netflix.spinnaker.orca.CancellableStage
+import com.netflix.spinnaker.orca.pipeline.LinearStage
+import com.netflix.spinnaker.orca.pipeline.model.Stage
+import groovy.util.logging.Slf4j
+import org.springframework.batch.core.Step
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Component
+
+@Slf4j
+@Component
+class AcaTaskStage extends LinearStage implements CancellableStage {
+  public static final String PIPELINE_CONFIG_TYPE = "acaTask"
+
+  @Autowired MonitorAcaTaskStage monitorAcaTaskStage
+
+  AcaTaskStage() {
+    super(PIPELINE_CONFIG_TYPE)
+  }
+
+  @Override
+  List<Step> buildSteps(Stage stage) {
+    Map canaryStageId = [canaryStageId: stage.id]
+
+    Map monitorContext = canaryStageId + stage.context
+
+    injectAfter(stage, "Monitor ACA Task", monitorAcaTaskStage, monitorContext)
```

Comment:
I was just following the pattern that the main canary stage was doing.  I’m open to a better way.

-- 
Zan Thrash
Sent with Airmail

On April 21, 2016 at 2:02:12 PM, Rob Fletcher (notifications@github.com) wrote:

In orca-mine/src/main/groovy/com/netflix/spinnaker/orca/mine/pipeline/AcaTaskStage.groovy:

> +class AcaTaskStage extends LinearStage implements CancellableStage {
> +  public static final String PIPELINE_CONFIG_TYPE = "acaTask"
> +
> +  @Autowired MonitorAcaTaskStage monitorAcaTaskStage
> +
> +  AcaTaskStage() {
> +    super(PIPELINE_CONFIG_TYPE)
> +  }
> +
> +  @Override
> +  List<Step> buildSteps(Stage stage) {
> +    Map canaryStageId = [canaryStageId: stage.id]
> +
> +    Map monitorContext = canaryStageId + stage.context
> +
> +    injectAfter(stage, "Monitor ACA Task", monitorAcaTaskStage, monitorContext)
Why is this done as a synthetic stage?

—
You are receiving this because you authored the thread.
Reply to this email directly or view it on GitHub